src/WebJobs.Script.WebHost/Models/MSIContext.cs

```diff
@@ -12,5 +12,11 @@ public class MSIContext
         public string MSISecret { get; set; }
 
         public IEnumerable<ManagedServiceIdentity> Identities { get; set; }
+
+        public ManagedServiceIdentity SystemAssignedIdentity { get; set; }
+
+        public IEnumerable<ManagedServiceIdentity> UserAssignedIdentities { get; set; }
```

Comment:
please see if there are existing unit tests. if yes, expand to include tests for new properties

Reply:
There are existing tests here: https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Management/InstanceManagerTests.cs#L571-L571

For the most part, MSI sidecar initialization is just a passthrough from perspective of Functions Host (serialize and forward payload). I think this should be by design to reduce added complexity of depending on MSIContext within Functions Host.

Test scenarios already covered:
1. If MSI endpoint doesn't exist
2. If MSIContext from DataRole is null
3. MSI sidecar needs no warmup ping
4. Successful specialization

Reply:
Added an additional test for properties that are expected during msi specialization. @pragnagopa